vms/platformvm/state/state.go

```diff
@@ -1766,21 +1765,16 @@ func (s *state) sync(genesis []byte) error {
 	return nil
 }
 
-func (s *state) init(genesisBytes []byte) error {
+func (s *state) init(g *genesis.Genesis) error {
 	// Create the genesis block and save it as being accepted (We don't do
 	// genesisBlock.Accept() because then it'd look for genesisBlock's
 	// non-existent parent)
-	genesisID := hashing.ComputeHash256Array(genesisBytes)
-	genesisBlock, err := block.NewApricotCommitBlock(genesisID, 0 /*height*/)
+	genesisBlock, err := block.NewApricotCommitBlock(g.GenesisID, 0 /*height*/)
```

Comment:
Instead of having genesis id as a field attached to Genesis, I think it would be cleaner to just hash the genesis file here and have the caller pass in the id. This way we don't have to have the `ID` field in `Genesis` which is only ever used here anyways.

Reply:
@joshua-kim I don't really have a file to has here, I have a byte slice. I understand your point about the ID being a block property.
I have replace the GenesisID attribute with GenesisBytes one and hashed it to get the genesis block ID wherever needed.
LMK if this is acceptable